core/src/service/file_metadata_service.rs

```diff
@@ -1,140 +1,392 @@
 use std::marker::PhantomData;
 
-use crate::client;
-use crate::client::{Client, GetUpdatesRequest};
-use crate::error;
-use crate::error_enum;
-use crate::model::file_metadata::FileMetadata;
+use crate::client::{
+    ChangeFileContentRequest, Client, CreateFileRequest, GetFileRequest, GetUpdatesRequest,
+};
+use crate::model::file_metadata::{FileMetadata, Status};
 use crate::repo;
 use crate::repo::account_repo::AccountRepo;
 use crate::repo::db_provider;
 use crate::repo::file_metadata_repo::FileMetadataRepo;
-use crate::API_LOC;
+use crate::repo::file_repo::FileRepo;
+use crate::service::file_encryption_service;
+use crate::service::file_encryption_service::FileEncryptionService;
+use crate::{client, debug, error, error_enum, info};
 use sled::Db;
 
 error_enum! {
     enum Error {
         ConnectionFailure(db_provider::Error),
         RetrievalError(repo::account_repo::Error),
         ApiError(client::ClientError),
+        FileRepoError(repo::file_repo::Error),
         MetadataRepoError(repo::file_metadata_repo::Error),
+        SerderError(serde_json::error::Error),
+        FileCreateError(file_encryption_service::FileCreationError),
     }
 }
 
 pub trait FileMetadataService {
-    fn update(db: &Db) -> Result<Vec<FileMetadata>, Error>;
+    fn sync(db: &Db) -> Result<Vec<FileMetadata>, Error>;
```

Comment:
Don't group `Error`'s like this. If these return different `Error`s expose that to the caller

Reply:
subsequent PR